tests/unit/runtime/comm/test_coalesced_collectives.py

```diff
@@ -59,3 +59,32 @@ def test(self):
             assert torch.allclose(output, torch.zeros_like(output))
         elif dist.get_rank() == 1:
             assert output.shape == (0, )
+
+
+# Currently we cannot test all_to_all_quant_reduce in non-fallback cases because we don't support multinodes tests.
+class TestAllToAllQuantReduceFallback(DistributedTest):
+    world_size = 2
+
+    def test_1d_tensor(self):
+        # case 1: 1D tensor
+        input = torch.zeros((10, ), dtype=torch.half, device=get_accelerator().current_device_name())
+        output = all_to_all_quant_reduce([input], {})[0]
+
+        if dist.get_rank() == 0:
+            assert output.shape == (5, )
+            assert torch.allclose(output, torch.zeros_like(output))
+        elif dist.get_rank() == 1:
+            assert output.shape == (5, )
+            assert torch.allclose(output, torch.zeros_like(output))
+
```

Comment:
I think we also need a test function for 2d tensor, which really calls 2 step all-to-all

Reply:
discussed offline. currently our tests cannot do multi nodes testing